src/app/components/leftbar/leftbar.component.ts

```diff
@@ -0,0 +1,17 @@
+import { Component, EventEmitter, Output } from '@angular/core';
+
+@Component({
+  selector: 'ccf-leftbar',
+  templateUrl: './leftbar.component.html',
+  styleUrls: ['./leftbar.component.scss']
+})
+export class LeftbarComponent {
+
+  @Output() drawerToggle = new EventEmitter<boolean>();
```

Comment:
This should be typed as Observable!

Reply:
do you mean use Observable instead of EventEmitter?

Reply:
You can use eventemitter under the hood, but the exposed type should be Observable.
Ex:
@Output() foo: Observable<>;
private readonly _foo = new EventEmitter;
constructor() {
  this.foo = this._foo.asObservable();
}

Reply:
is there any secific reason for this, or any code guide that you can point me to that can justify the usage of observables as @Output vs event emitters.

Reply:
The issue with making EventEmitters accessible outside of the component is that then something else can trigger those events and hijack the system. Maybe for mundane reasons, but then it is a security and stability concern.